schemagen/src/main/java/gov/nist/secauto/metaschema/schemagen/json/impl/AbstractModelDefinitionJsonSchema.java

```diff
@@ -71,7 +71,13 @@ protected AbstractModelDefinitionJsonSchema(
 
     // determine the flag instances to generate
     if (jsonKeyFlagName != null) {
-      IFlagInstance jsonKeyFlag = ObjectUtils.requireNonNull(definition.getFlagInstanceByName(jsonKeyFlagName));
+      IFlagInstance jsonKeyFlag = definition.getFlagInstanceByName(jsonKeyFlagName);
+      if (jsonKeyFlag == null) {
+        throw new IllegalArgumentException(
+            String.format("The referenced JSON key flag '%s' does not exist on definition '%s'.",
```

Comment:
Minor, non-blocking: given the above similar feedback maybe?

```suggestion
            String.format("The referenced json-key flag-name '%s' does not exist on definition '%s'.",
```